src/common/utils/IndexKeyUtils.cpp

```diff
@@ -156,25 +157,58 @@ Value IndexKeyUtils::parseIndexTTL(const folly::StringPiece& raw) {
 }
 
 // static
-StatusOr<std::vector<std::string>> IndexKeyUtils::collectIndexValues(
-    RowReader* reader, const std::vector<nebula::meta::cpp2::ColumnDef>& cols) {
+StatusOr<std::vector<std::string>>
+IndexKeyUtils::collectIndexValues(RowReader* reader,
+                                  const std::vector<nebula::meta::cpp2::ColumnDef>& cols,
+                                  const meta::SchemaProviderIf* latestSchema) {
   if (reader == nullptr) {
     return Status::Error("Invalid row reader");
   }
   std::vector<Value> values;
   for (const auto& col : cols) {
-    auto v = reader->getValueByName(col.get_name());
+    auto propName = col.get_name();
+    auto val = readValueWithLatestSche(reader, propName, latestSchema);
+    if (!val.ok()) {
+      LOG(ERROR) << "prop error by : " << propName << ". status : " << val.status();
+      return val.status();
+    }
+    auto v = val.value();
     auto isNullable = col.nullable_ref().value_or(false);
     auto ret = checkValue(v, isNullable);
     if (!ret.ok()) {
-      LOG(ERROR) << "prop error by : " << col.get_name() << ". status : " << ret;
+      LOG(ERROR) << "prop error by : " << propName << ". status : " << ret;
       return ret;
     }
     values.emplace_back(std::move(v));
   }
   return encodeValues(std::move(values), cols);
 }
 
+// static
+StatusOr<Value>
+IndexKeyUtils::readValueWithLatestSche(RowReader* reader,
+                                       const std::string propName,
+                                       const meta::SchemaProviderIf* latestSchema) {
+  auto value = reader->getValueByName(propName);
+  if (latestSchema == nullptr
```

Comment:
Perhaps there is one more condition here to check if the value is a "Null" (literally, sorry about the naming). You only handle the UNKNOWN_PROP.

See the code in RowReader
![image](https://user-images.githubusercontent.com/13706157/143370260-38ece1e6-9025-45fb-aaca-6eb9bd6c03b7.png)

Reply:
The condition NullType::__NULL__ is handled by `IndexKeyUtils::checkValue`. Maybe I don't need to handle it in function  `readValueWithLatestSche`.

Reply:
I mean if the value in RowReader is the `NullType::NULL`, we should directly return it. What do you think?

Reply:
Ah, my bad, I see the point here. Never mind.